File: programmers/lv0/코딩기초트레이닝/조건문/원소들의곱과합/Solution.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> num_list) {
    int multiplyAll = 1;
    int sum = 0;
    for (int i : num_list) {
        multiplyAll *= i;
        sum += i;
    }
    return multiplyAll < sum * sum ? 1 : 0;
}

File: programmers/lv0/코딩기초트레이닝/조건문/원소들의곱과합/Solution.js
/**
 * @param {int[]}num_list
 * @returns {number}
 */
const solution = (num_list) => {
    const mul = num_list.reduce((acc, cur) => acc * cur, 1)
    const sum = num_list.reduce((acc, cur) => acc + cur, 0)
    return mul < sum ** 2 ? 1 : 0
}

const x = solution([3, 4, 5, 2, 1])
const y = solution([5, 7, 8, 3])
console.info('x', x)
console.info('y', y)